Add custom lt_glow textures support for player and traffic vehicles

Wanted9811 opened the pull request:
This feature allows vehicles to use their own custom lt_glow textures if they're exist otherwise they will use the default lt_glow texture.

Texture names (Player):
"lt_glow_siren" = custom siren texture.
"lt_glow_headlight" = custom headlight texture.

Texture name (Traffic):
"lt_glow_ambient" = custom headlight texture.

Wanted9811 committed:
Add custom lt_glow textures support for player and traffic vehicles

This feature allows vehicles to use their own custom lt_glow textures if they're exist otherwise it will use the default lt_glow texture.

Texture names (Player):
"lt_glow_siren" = custom siren texture.
"lt_glow_headlight" = custom headlight texture.

Texture name (Traffic):
"lt_glow_ambient" = custom headlight texture.

0x1F9F1 commented on src/handlers/feature_handlers.cpp:
```diff
@@ -5201,6 +5201,25 @@ void vehCarModelFeatureHandler::DrawHeadlights(bool rotate)
+    gfxTexture* headlightTexture;
+
+    //first time texture load
+    if (!glowLoaded) {
```
`glowLoaded` and `headlightTexture` aren't static, so this condition will always be true (note I also wouldn't recommend making them static inside the function, because that would cause issues between races).

It would be much better to load/free them at the same time as the original `ltLight::GlowTexture` (inside `ltLight::ltLight` and `ltLight::ShutdownLights`).

Wanted9811 replied:
The condition should be always true to avoid loading the texture every time which means this should be loaded once. I don't think there's something wrong with this, the changes you wanna me to do can't be made because it's not easy to do as you think. If you check out `dgBangerInstance::DrawGlow` inside the hook, you will find that it had been implemented in a similar way setting up the red glow texture to be used for banger glows.

0x1F9F1 replied:
Always true, as in the if statement always executes. You are setting `glowLoaded = false;`, then immediately checking `!glowLoaded`, which will always evaluate to true.

Wanted9811 committed:
Fix rich presence crash that happens when it's disabled

That was happening because of the Red glow texture which was being loaded in a wrong way, now we fixed this via loading it in `ltLight` constructor instead and free it in `ShutdownLights`, we also did the same thing to the custom lt glow textures.

0x1F9F1 commented on src/handlers/feature_handlers.cpp:
```diff
@@ -4054,6 +4055,92 @@ void ltLensFlareHandler::Install() {
+{
+    if (ltLight::GlowTexture)
+    {
+        gfxFreeTexture((gfxTexture*)ltLight::GlowTexture);
```
Unncessary cast?

Wanted9811 replied:
It must be there to convert its type properly otherwise the game will crash on freeing it so the cast is necessary.
